src/lxc/tools/lxc_unshare.c

```diff
@@ -41,10 +41,112 @@
 
 #include "arguments.h"
 #include "caps.h"
-#include "conf.h"
+#include "log.h"
 #include "namespace.h"
 #include "utils.h"
 
+lxc_log_define(lxc_unshare, lxc);
+
+struct start_arg {
+	char *const *args;
+	int flags;
+	uid_t uid;
+	bool setuid;
+	int want_default_mounts;
+	int wait_fd;
+	const char *want_hostname;
+};
+
+struct my_iflist
+{
+	char *mi_ifname;
+	struct my_iflist *mi_next;
+};
+
+static int my_parser(struct lxc_arguments *args, int c, char *arg);
+static inline int sethostname_including_android(const char *name, size_t len);
+static int get_namespace_flags(char *namespaces);
+static bool lookup_user(const char *optarg, uid_t *uid);
+static int mount_fs(const char *source, const char *target, const char *type);
+static void lxc_setup_fs(void);
+static int do_start(void *arg);
+
+static struct my_iflist *tmpif, *my_iflist = NULL;
+
+static const struct option my_longopts[] = {
+	{"namespaces", required_argument, 0, 's'},
+	{"user", required_argument, 0, 'u'},
+	{"hostname", required_argument, 0, 'H'},
+	{"ifname", required_argument, 0, 'i'},
+	{"daemon", no_argument, 0, 'd'},
+	{"remount", no_argument, 0, 'M'},
+	LXC_COMMON_OPTIONS
+};
+
+static struct lxc_arguments my_args = {
+	.progname     = "lxc-unshare",
+	.help         = "\
+-s NAMESPACES COMMAND\n\
+\n\
+lxc-unshare run a COMMAND in a new set of NAMESPACES\n\
+\n\
+Options :\n\
+  -s, --namespaces=FLAGS\n\
+                    ORed list of flags to unshare:\n\
+                    MOUNT, PID, UTSNAME, IPC, USER, NETWORK\n\
+  -u, --user=USERID\n\
+                    new id to be set if -s USER is specified\n\
+  -H, --hostname=HOSTNAME\n\
+                    Set the hostname in the container\n\
+  -i, --ifname=IFNAME\n\
+                    Interface name to be moved into container (presumably with NETWORK unsharing set)\n\
+  -d, --daemon      Daemonize (do not wait for container to exit)\n\
+  -M, --remount     Remount default fs inside container (/proc /dev/shm /dev/mqueue)\n\
+",
+	.options      = my_longopts,
+	.parser       = my_parser,
+	.checker      = NULL,
+	.daemonize    = 0,
+	.pidfile      = NULL,
+};
+
+static int my_parser(struct lxc_arguments *args, int c, char *arg)
+{
+	switch (c) {
+	case 's':
+		args->flags = get_namespace_flags(arg);
+		if (args->flags < 0)
+			return -1;
+		break;
+	case 'u':
+		if (!lookup_user(arg, &args->uid))
+			return -1;
+
+		args->setuid = true;
+		break;
+	case 'H':
+		args->want_hostname = arg;
+		break;
+	case 'i':
+		if (!(tmpif = malloc(sizeof(*tmpif)))) {
```

Comment:
Please do:
```
tmpif = malloc(sizeof(*tmpif));
if (!tmpif) {
        /* handle error */
}
```
:)

Reply:
@brauner I used existing codes. So, I didn't check it. Thanks for your review.